modules/beta-private-cluster/firewall.tf

```diff
@@ -196,3 +197,50 @@ resource "google_compute_firewall" "shadow_allow_nodes" {
     metadata = "INCLUDE_ALL_METADATA"
   }
 }
+
+resource "google_compute_firewall" "shadow_allow_inkubelet" {
+  count = var.add_shadow_firewall_rules ? 1 : 0
+
+  name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-inkubelet"
+  description = "Managed by terraform GKE module: A shadow firewall rule to match the default rule allowing worker nodes & pods communication to kubelet."
+  project     = local.network_project_id
+  network     = var.network
+  priority    = min(998, var.shadow_firewall_rules_priority) # rule created by GKE robot have prio 999
+  direction   = "INGRESS"
+
+  source_ranges = local.pod_all_ip_ranges
+  source_tags   = [local.cluster_network_tag]
+  target_tags   = [local.cluster_network_tag]
+
+  allow {
+    protocol = "tcp"
+    ports    = ["10255"]
+  }
+
+  log_config {
+    metadata = "INCLUDE_ALL_METADATA"
+  }
+}
+
+resource "google_compute_firewall" "shadow_deny_exkubelet" {
+  count = var.add_shadow_firewall_rules ? 1 : 0
+
+  name        = "gke-shadow-${substr(var.name, 0, min(25, length(var.name)))}-exkubelet"
+  description = "Managed by terraform GKE module: A shadow firewall rule to match the default deny rule to kubelet."
+  project     = local.network_project_id
+  network     = var.network
+  priority    = min(999, var.shadow_firewall_rules_priority) # rule created by GKE robot have prio 1000
```

Comment:
Do we want to validate in variable "validation" argument?

variable "firewall_priority" {
  type        = number
  description = "Priority rule for firewall rules"
  default     = 1000

  validation {
    condition     = var.firewall_priority < 1000
    error_message = "The shadow firewall rule priority must be lower than auto-created one, i.e. 1000."
 }

Reply:
Good point. Added.